File: webshared/src/main/scala/com/taskadapter/web/ConnectorSetupPanel.scala
package com.taskadapter.web

import com.taskadapter.connector.definition.ConnectorSetup
import com.vaadin.flow.component.Component

trait ConnectorSetupPanel {
  def getUI: Component

  /**
    * @return None if no errors. localized error text otherwise
    */
  def validate: Option[String]

  def showError(String: String) : Unit

  def getResult: ConnectorSetup
}

File: webshared/src/main/java/com/taskadapter/web/ConnectorSetupPanel.java
package com.taskadapter.web;

import com.taskadapter.connector.definition.ConnectorSetup;
import com.vaadin.flow.component.Component;

import java.util.Optional;

public interface ConnectorSetupPanel {
    Component getComponent();

    /**
     * @return Empty if no errors. localized error text otherwise
     */
    Optional<String> validate();

    void showError(String string);

    ConnectorSetup getResult();
}
